zerver/lib/send_email.py

```diff
@@ -514,6 +516,14 @@ def get_header(option: Optional[str], header: Optional[str], name: str) -> str:
     return str(option or header)
 
 
+def parse_email_template(template: str) -> Tuple[Message, str]:
+    with open(template) as f:
+        text = f.read()
+        message = Parser(policy=default).parsestr(text)
+        hash_code = hashlib.sha256(text.encode()).hexdigest()[0:32]
+        return message, hash_code
```

Comment:
Extracting this function seems like a good refactor, but you should do it in its own preparatory commit, not squashed with your actual changes.

Check out our [commit discipline guide](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for more details.

Reply:
This first commit should have a commit message explaining these changes, not a duplicate of that for the main PR.

Read a dozen random recently merged PRs and you'll see some good examples.